set.mm

```diff
@@ -742983,23 +742983,196 @@ to Davis and Putnam (1960).  (Contributed by David A. Wheeler,
+    ( wi impsingle-step15 impsingle-step20 ax-mp ) CABDZDHDCBDZHDZDHCDCDJDCHABE
+    CHICFG $.
+
+  $( Derivation of impsingle-step22 from ~ ax-mp and ~ impsingle .  It is used
+     as a lemma in proofs of ~ imim1 and ~ peirce from ~ impsingle .  It is
+     Step 22 in Lukasiewicz, where it appears as 'Cpp' using parenthesis-free
+     prefix notation.  (Contributed by Larry Lesyna and Jeffrey P. Machado,
+     2-Aug-2023.)  (Proof modification is discouraged.)
+     (New usage is discouraged.)) $)
+  impsingle-step22 $p  |- ( ph -> ph ) $=
+    ( wth wmu wla wps wi impsingle-step4 impsingle ax-mp ) BCFBFDBFFZAAFZBCDGAE
+    FZAFZKFZJKFZAEAGKAFMFNOFAALGKAMJHIII $.
+
+  $( Derivation of impsingle-step25 from ~ ax-mp and ~ impsingle .  It is used
+     as a lemma in the proof of ~ imim1 from ~ impsingle .  It is Step 25 in
+     Lukasiewicz, where it appears as 'CCpqCCCprqq' using parenthesis-free
+     prefix notation.  (Contributed by Larry Lesyna and Jeffrey P. Machado,
+     2-Aug-2023.)  (Proof modification is discouraged.)
+     (New usage is discouraged.) $)
+  impsingle-step25 $p  |- ( ( ph -> ps )
+                         -> ( ( ( ph -> ch ) -> ps ) -> ps ) ) $=
+    ( wth wi impsingle-step22 impsingle-step20 impsingle-step8 impsingle-step15
+    ax-mp ) ACEZKBEZBEZEZABEMEBDEZKEMEZNMMEPMFKBLDGJOKMHJACLBIJ $.
+
+  $( Derivation of impsingle-imim1 ( ~ imim1 ) from ~ ax-mp and ~ impsingle .
+     It is step 29 in Lukasiewicz.  (Contributed by Larry Lesyna and Jeffrey P.
+     Machado, 2-Aug-2023.)  (Proof modification is discouraged.)
+     (New usage is discouraged.) $)
+  impsingle-imim1 $p |- ( ( ph -> ps )
+                          -> ( ( ps -> ch ) -> ( ph -> ch ) ) ) $=
+    ( wi impsingle-step21 impsingle-step25 ax-mp ) ACDZBDBDZBCDHDZDZABDZJDZACBE
+    MIDIDZKMDLIDNABCFLIJFGLJIEGG $.
+
+  $( Derivation of impsingle-peirce ( ~ peirce ) from ~ ax-mp and ~ impsingle .
+     It is step 28 in Lukasiewicz.  (Contributed by Larry Lesyna and Jeffrey P.
+     Machado, 2-Aug-2023.)  (Proof modification is discouraged.)
+     (New usage is discouraged.) $)
+  impsingle-peirce $p |- ( ( ( ph -> ps ) -> ph ) -> ph ) $=
+    ( wi impsingle-step22 impsingle-step25 ax-mp ) AACABCACACADAABEF $.
+
+  $( Derivation of ~ ax-2 from the Tarski-Bernays axiom set { ~ ax-1 ,
+     ~ imim1 , ~ peirce }.  Note that no inference rule other than ~ ax-mp is
+     used in this proof.  This proof establishes a circle of equivalence:  From
+     { ~ impsingle }, { ~ ax-1 , ~ imim1 , ~ peirce } was proved.  From {
+     ~ ax-1 , ~ imim1 , ~ peirce }, { ~ ax-1 , ~ ax-2 and ~ peirce } was
+     proved.  From { ~ ax-1 , ~ ax-2 and ~ peirce }, { ~ impsingle } was
+     proved.  Therefore, the theorems that can be proved by selecting any one
+     of these three distinct axiom sets must be equivalent.  Prover9 and mmj2
+     were used to help constuct this proof.  (Contributed by Larry Lesyna and
+     Jeffrey P. Machado, 1-Aug-2023.)  (New usage is discouraged.) $)
```

Comment:
It would be safer to add "proof modif discouraged" since an automatic minimization could simply rewrite this proof to use ~ax-2 itself.

Reply:
done